bots/elasticsearch/tasks.py

```diff
@@ -37,50 +56,63 @@ def bulk_stream(self, model, ids):
 
     def serialize(self, inst):
         return {
-            Person: self.serialize_person,
             AbstractCreativeWork: self.serialize_creative_work,
+            Entity: self.serialize_entity,
+            Person: self.serialize_person,
+            Tag: self.serialize_tag,
         }[type(inst)._meta.concrete_model](inst)
 
-    def serialize_person(self, person):
-        return {
+    def serialize_person(self, person, suggest=True):
+        serialized_person = {
             '@id': person.pk,
             '@type': 'person',
             'suffix': safe_substr(person.suffix),
             'given_name': safe_substr(person.given_name),
             'family_name': safe_substr(person.family_name),
-            'full_name': safe_substr(person.get_full_name()),
+            'name': safe_substr(person.get_full_name()),
             'additional_name': safe_substr(person.additional_name),
             'identifiers': [{
                 'url': identifier.url,
                 'base_url': identifier.base_url,
             } for identifier in person.identifiers.all()],
             'affiliations': [
-                self.serialize_entity(affiliation)
+                self.serialize_entity(affiliation, False)
                 for affiliation in
                 person.affiliations.all()
             ],
             'sources': [source.robot for source in person.sources.all()],
         }
+        return add_suggest(serialized_person) if suggest else serialized_person
 
-    def serialize_entity(self, entity):
-        return {
+    def serialize_entity(self, entity, suggest=True):
+        serialized_entity = {
             '@id': entity.pk,
-            'name': safe_substr(entity.name),
             '@type': type(entity).__name__.lower(),
+            'name': safe_substr(entity.name),
+            'url': entity.url,
+            'location': safe_substr(entity.location),
+        }
+        return add_suggest(serialized_entity) if suggest else serialized_entity
+
+    def serialize_tag(self, tag, suggest=True):
+        serialized_tag = {
+            '@id': str(tag.pk),
+            '@type': 'tag',
+            'name': safe_substr(tag.name),
         }
+        return add_suggest(serialized_tag) if suggest else serialized_tag
 
     def serialize_creative_work(self, creative_work):
         return {
             '@type': type(creative_work).__name__.lower(),
-            'associations': [
-                self.serialize_entity(association)
-                for association in [
-                    *creative_work.funders.all(),
-                    *creative_work.publishers.all(),
-                    *creative_work.institutions.all(),
-                    *creative_work.organizations.all()
-                ]
-            ],
+            'funders': [self.serialize_entity(entity, False)
```

Comment:
Please leave all of these on a single line, flake8 is set to ignore line length for this project.